spec/index.md

```diff
@@ -2470,6 +2470,50 @@ This function returns fee percentiles, measured in millisatoshi/vbyte (1000 mill
 
+### IC method `take_canister_snapshot` {#ic-take_canister_snapshot}
+
+This method takes a snapshot of the specified canister. A snapshot consists of the wasm memory, stable memory, certified variables, wasm chunk store and wasm binary.
+
+Subsequent `take_canister_snapshot` calls will create a new snapshot. However, a `take_canister_snapshot` call might fail if the maximum number of snapshots per canister is reached. This error can be avoided by providing a snapshot ID via the optional `replace_snapshot` parameter. The snapshot identified by the specified ID will be deleted once a new snapshot has been successfully created. Currently, only one snapshot per canister is allowed.
+
+It's important to note that a snapshot will increase the memory footprint of the canister. Thus, the canister's balance must have a sufficient amount of cycles to support the new freezing threshold.
+
+Only controllers can take a snapshot of a canister and load it back to the canister.
+
+:::note
+
+It's important to stop a canister before taking a snapshot to ensure that all outstanding callbacks are completed. Failing to do so may cause the canister to not make sense of the callbacks if its state is restored using the snapshot.
+It is expected that the canister controllers (or their tooling) do this separately.
+
+:::
+
+### IC method `load_canister_snapshot` {#ic-load_canister_snapshot}
+
+This method loads a snapshot identified by `snapshot_id` onto the canister. It fails if no snapshot with the specified `snapshot_id` can be found.
+
+Only controllers can take a snapshot of a canister and load it back to the canister.
+
+:::note
+
+It's important to stop a canister before loading a snapshot to ensure that all outstanding callbacks are completed. Failing to do so may cause the canister to not make sense of the callbacks if its state is restored.
+It is expected that the canister controllers (or their tooling) do this separately.
+
+:::
+
+The optional `sender_canister_version` parameter can contain the caller's canister version. If provided, its value must be equal to `ic0.canister_version`.
+
+### IC method `list_canister_snapshots` {#ic-list_canister_snapshots}
+
+This method lists the snapshots of the canister identified by `canister_id`. Only controllers of the canister can list its snapshots. Currently, at most one snapshot per canister will be stored.
+
+### IC method `delete_canister_snapshot` {#ic-delete_canister_snapshot}
+
+This method deletes a specified snapshot that belongs to an existing canister. An error will be returned if the snapshot is not found. 
+
+A snapshot cannot be found if it was never created, it was previously deleted, replaced by a new snapshot through a `take_canister_snapshot` request, or if the canister itself has been deleted or run out of cycles.
```

Comment:
> run out of cycles

Just to double check, if the canister is out of cycles, we lose all the snapshots?

Reply:
Yes.

Reply:
Double check again, `uninstall_code` keeps the snapshot, only `delete_canister` removes the snapshot?

Reply:
The `uninstall_code` method is sort of reseting the canister state. The canister ID is still the same, so it makes sense to keep the snapshots. On the other hand, `delete_canister` method deletes the canister ID, so at the moment there is no need to keep the snapshots any longer.

This behaviors might change in the future though, when the snapshot load across canisters is supported...

Reply:
> This behaviors might change in the future though, when the snapshot load across canisters is supported...

Could you please elaborate on what exactly might change?

Reply:
It's not even scoped yet. However, sharing snapshots across canister might require some changes in snapshot ownership...